### DISCUSSION TITLE
wire:model.defer="variable" after second submit shows null

### DISCUSSION THREAD

**Discussion by SushilMore1988** [livewire/livewire#5077] [1. Help]
This is blade view
```
<x-jet-form-section submit="store">
    <div class="col-span-6 sm:col-span-4">
                <x-jet-label for="name" value="{{ __('Name') }}" />
                <x-jet-input name="name" wire:model.defer="name" type="text" class="mt-1 block w-full" autocomplete="off" />
                <x-jet-input-error for="name" class="mt-2" />
            </div>
</x-jet-form-section>
```

This livewire component function
```
public function store() {
  dd($this);
}
```

if you submit the form then it shows the value in name then you close dd and again submit the form it shows null. this is how i generated this. To test it in actual scenario I have form where I am validating data if this validation fails it return with error and if I correct the invalid data and submit the other fields are null.

**Comment by msonowal** [comment]
I am running to this same issues had completely rewrite the form and the issue exists.

info the form is a full page livewire component and on first submit its fine and if in case of some exception I will return error than again if we submit it shows null

**Comment by joshhanley** [comment]
@SushilMore1988 the issue is the `dd()`, try using `dump()` instead. The reason is, that `dd()` is dump and **Die**, which means Livewire doesn't finish the request properly, hence why you are getting null on the next request.

Hope this helps!

**Reply by SushilMore1988** [reply]
This is happening when we are validating the request. If there is error in validation then you correct the error and again click on submit then it appears.

**Reply by joshhanley** [reply]
@SushilMore1988 can you create a test component that has only a single input in it that replicates the problem and paste the component class and view here so we can test? Thanks!

**Reply by SushilMore1988** [reply]
```<div>
    <form wire:submit.prevent="store">
        <div class="col-span-6 sm:col-span-4">
            <label for="name">{{ __('Name') }}</label>
            <input name="name" wire:model.defer="name" type="text" class="mt-1 block w-full" autocomplete="off" />
            @error('name')
                {{ $message }}
            @enderror
        </div>
        <div class="col-span-6 sm:col-span-4">
            <label for="email">{{ __('Email') }}</label>
            <input name="email" wire:model.defer="email" type="email" class="mt-1 block w-full" autocomplete="off" />
            @error('email')
                {{ $message }}
            @enderror
         </div>
         <button type="submit" class="btn btn-primary">Submit</button>
    </form>
</div>```

And in component file
```
class Test extends Component{
public $name, $email;

public function store() {
     $this->validate([
                       'name' => 'required|max:20',
                       'email' => 'required|email'
                      ]);
     echo "Done";
}
}
```

**Reply by joshhanley** [reply]
@SushilMore1988 thanks! It looks like half the html is missing and there are no properties in the component class. Can you provide everything I need to copy and paste straight into a component?

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

**Comment by jtrod** [comment]
I'm having the same issue. In my case, it is happening on a child component. Parent components keeps state, but child looses data and sets to property default on second request. But it works if I remove .defer or If I use .debounce instead

**Reply by joshhanley** [reply]
@jtrod make sure you have `wire:key`'s on your child components otherwise they may loose state. If that doesn't work, can you share your code so we can have a look?
